summarize_pr.py

```diff
@@ -42,29 +41,162 @@ def summarize_pr(
+        response_mime_type="application/json",
+        max_output_tokens=8192,
+    )
+
+    review_json_format = "{'body': 'Summary of the review', 'comments': ['path': 'The relative path to the file that necessitates a review comment.', 'body': 'The content of the review comment.', 'line': '(int) The line of the diff that the comment applies to.', 'side': 'RIGHT']}"
+
+    prompt = [
+        "The following is a git diff of a GitHub Pull Request.",
+        "Your task is to complete a thorough code review of the GitHub Pull Request. Analyze each code file line by line, then list any bugs you find and improvements that can be made to the code quality for readability and maintainability. Provide output in this JSON structure.",
+        review_json_format,
+        "---Git Diff---",
+        pr_diff,
+    ]
+
+    print("---Prompt---\n", prompt)
+    response = model.generate_content(prompt, generation_config=generation_config)
+    print("---Gemini Response---\n", response)
+
+    return response.text
+
+
+def convert_to_sarif(model_id: str, analysis_content: str) -> str:
+    model = GenerativeModel(
+        model_id,
+        system_instruction=[
+            "You are an expert software engineer.",
+        ],
+    )
+
+    generation_config = GenerationConfig(
+        response_mime_type="application/json",
+        max_output_tokens=8192,
     )
 
+    sarif_schema = requests.get(
+        "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/123e95847b13fbdd4cbe2120fa5e33355d4a042b/Schemata/sarif-schema-2.1.0.json"
+    ).text
+
+    prompt = [
+        "The following is analysis of a GitHub Pull Request.",
+        "Your task is to convert this into SARIF (Static Analysis Results Interchange Format) format.",
+        "This is the SARIF schema:",
+        sarif_schema,
+        "Analysis Content:",
+        analysis_content,
+        "SARIF:"
+        """{"$schema": "https://json-schema.org/draft/2020-12/schema#", "version": "2.1.0", "runs": [{"tool": {"driver": {"name": "Gemini", "shortDescription": {"text": "A large language model capable of generating text, translating languages, writing different kinds of creative content, and answering your questions in an informative way."}, "fullDescription": {"text": "Gemini Pro is a large language model (LLM) developed by Google AI. It is trained on a massive dataset of text and code, and can be used for a wide variety of tasks, including text generation, translation, writing different kinds of creative content, and answering your questions in an informative way."}}}, "results":""",
+    ]
+
+    print("---Prompt---\n", prompt)
+    response = model.generate_content(prompt, generation_config=generation_config)
+    print("---Gemini Response---\n", response)
+
+    return response.text
+
+
+def upload_sarif(api: GhApi, pr_number: str, sarif_text: str):
+    encoded_data = base64.b64encode(gzip.compress(sarif_text.encode("utf-8"))).decode(
+        "utf-8"
```

Comment:
The function is now responsible for generating a SARIF file, which is useful for storing code analysis results. However, it is important to ensure that the SARIF file follows the schema correctly. It might be beneficial to validate the generated SARIF against the schema to ensure its accuracy and compliance.